### PR TITLE
Add missing fields to transactions list, new psp reference design

### PR DESCRIPTION
**Pull request by witoszekdev**
I want to merge this change because it adds new fields to transactions list added in the Core PR, it also adds new design for psp reference column in Transactions' events list

## TODO
- [x] Add psp reference copy / link

**PR intended to be tested with API branch:** 

### Screenshots
CleanShot 2022-12-15 at 14 25 59@2x

### Pull Request Checklist

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `[data-test-id]` are added for new elements
4. [ ] Changes are mentioned in the changelog
5. [ ] The changes are tested in different browsers and in light/dark mode

### Test environment config

API_URI=https://automation-dashboard.staging.saleor.cloud/graphql/
MARKETPLACE_URL=https://apps.saleor.io
SALEOR_APPS_ENDPOINT=https://apps.saleor.io/api/saleor-apps

### Do you want to run more stable tests?
To run all tests, just select the stable checkbox. To speed up tests, increase the number of containers. Tests will be re-run only when the "run e2e" label is added.

1. [ ] stable
2. [ ] giftCard
3. [ ] category
4. [ ] collection
5. [ ] attribute
6. [ ] productType
7. [ ] shipping
8. [ ] customer
9. [ ] permissions
10. [ ] menuNavigation
11. [ ] pages
12. [ ] sales
13. [ ] vouchers
14. [ ] homePage
15. [ ] login
16. [ ] orders
17. [ ] products
18. [ ] app
19. [ ] plugins
20. [ ] translations
21. [ ] navigation
22. [ ] variants
23. [ ] payments

CONTAINERS=1

### CODE REVIEW

**Comment by witoszekdev** [comment]
@Droniu @poulch please ignore failing tests - there's some issue with Polly.js 🤷🏻‍♂️ 

I'll try to fix this on the feature branch, promise ;)

File: src/orders/components/OrderTransaction/components/TransactionEvents/components/EventType.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { TransactionEventFragment } from "@saleor/graphql";
+import { transactionEventTypeMap } from "@saleor/orders/messages";
+import React from "react";
+import { FormattedMessage } from "react-intl";
+
+export const EventType = ({ event }: { event: TransactionEventFragment }) => {
```
Comment (poulch) [inline src/orders/components/OrderTransaction/components/TransactionEvents/components/EventType.tsx:6]:
```suggestion
interface EventTypeProps  { 
    event: TransactionEventFragment 
}

export const EventType = ({ event }: EventTypeProps) => {
```

File: src/orders/components/OrderTransaction/components/TransactionEvents/components/PspReference.tsx [hunk 2]
```diff
@@ -1,101 +1,91 @@
...
-  }, [renderStep]);
-
-  useEffect(() => {
-    if (ref.current && renderStep === RenderStep.final) {
-      // Forces re-calculation of width when text changes
-      setWidth(null);
-      setRenderStep(RenderStep.initial);
-    }
-  }, watch);
+const ReferenceLink = ({ href, children }) => {
```
Comment (poulch) [inline src/orders/components/OrderTransaction/components/TransactionEvents/components/PspReference.tsx:49]: Missing types for props

File: src/orders/components/OrderTransaction/components/TransactionEvents/components/PspReference.tsx [hunk 3]
```diff
@@ -1,101 +1,91 @@
...
+      </a>
+    );
+  }
 
-  return { ref: onMount, width, renderStep };
+  return children;
 };
 
-export const PspReference: React.FC<{ reference: string }> = ({
+export const PspReference: React.FC<{ reference: string; url?: string }> = ({
```
Comment (poulch) [inline src/orders/components/OrderTransaction/components/TransactionEvents/components/PspReference.tsx:61]: I'd would extract type into interface `PspReferenceProps`

File: src/orders/components/OrderTransactionPayment/utils.ts [hunk 4]
```diff
@@ -31,7 +32,51 @@ export const findMethodName = (
 ): string =>
   allMethods.find(method => method.id === gatewayId)?.name ?? gatewayId;
 
-export const mapTransactionsToEvents = (
+const mapPaymentKindToTransactionType = (
+  kind: TransactionKind,
+): [TransactionEventActionTypeEnum | null, TransactionEventStatus | null] => {
```
Comment (poulch) [inline src/orders/components/OrderTransactionPayment/utils.ts:37]:
```suggestion
): [TransactionEventActionTypeEnum, TransactionEventStatus] | [null, null] => {
```

**Review by Droniu — APPROVED** [review]
LGTM

**Pull request merged**